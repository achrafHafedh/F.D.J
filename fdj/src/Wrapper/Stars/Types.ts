export type StarsProps = {
  dispatch: Function;
};
