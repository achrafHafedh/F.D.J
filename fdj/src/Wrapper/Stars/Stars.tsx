import { FunctionComponent } from "react";
import { StarsProps } from "./Types";
import Star from "../../Components/Star";
import { STARS_LENGTH } from "../../Config";

import "./Stars.scss";

const Stars: FunctionComponent<StarsProps> = ({ dispatch }) => {
  const renderStars = () => {
    const listStar = [];
    for (let index = 0; index < STARS_LENGTH; index++) {
      listStar.push(
        <Star key={index} dispatch={dispatch} className={index >= 9 ? "big" : ""} index={index} value={index + 1} />
      );
    }
    return listStar;
  };

  return <div className="container-numbers">{renderStars()}</div>;
};

export default Stars;
