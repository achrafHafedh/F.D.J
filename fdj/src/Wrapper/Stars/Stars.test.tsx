import { render } from "@testing-library/react";

import Stars from "./Stars";

describe("<Stars/>", () => {
  const props = {
    dispatch: () => "",
  };
  it("should render Stars component", () => {
    const { container } = render(<Stars {...props} />);
    expect(container).toMatchSnapshot();
  });
});
