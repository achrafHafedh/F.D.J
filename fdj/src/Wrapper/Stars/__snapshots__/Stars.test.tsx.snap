// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stars/> should render Stars component 1`] = `
<div>
  <div
    class="container-numbers"
  >
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-0"
        type="checkbox"
        value="1"
      />
      <label
        class=""
        for="star-0"
      >
        1
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-1"
        type="checkbox"
        value="2"
      />
      <label
        class=""
        for="star-1"
      >
        2
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-2"
        type="checkbox"
        value="3"
      />
      <label
        class=""
        for="star-2"
      >
        3
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-3"
        type="checkbox"
        value="4"
      />
      <label
        class=""
        for="star-3"
      >
        4
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-4"
        type="checkbox"
        value="5"
      />
      <label
        class=""
        for="star-4"
      >
        5
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-5"
        type="checkbox"
        value="6"
      />
      <label
        class=""
        for="star-5"
      >
        6
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-6"
        type="checkbox"
        value="7"
      />
      <label
        class=""
        for="star-6"
      >
        7
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-7"
        type="checkbox"
        value="8"
      />
      <label
        class=""
        for="star-7"
      >
        8
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-8"
        type="checkbox"
        value="9"
      />
      <label
        class=""
        for="star-8"
      >
        9
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-9"
        type="checkbox"
        value="10"
      />
      <label
        class="big"
        for="star-9"
      >
        10
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-10"
        type="checkbox"
        value="11"
      />
      <label
        class="big"
        for="star-10"
      >
        11
      </label>
      <div
        class="star"
      />
    </div>
    <div
      class="container-star"
    >
      <input
        class="visually-hidden"
        id="star-11"
        type="checkbox"
        value="12"
      />
      <label
        class="big"
        for="star-11"
      >
        12
      </label>
      <div
        class="star"
      />
    </div>
  </div>
</div>
`;
